packages/toolkit/runtime-utils/src/browser/nestedRoutes.tsx

```diff
@@ -131,6 +139,25 @@ function createLoader(route: NestedRoute): LoaderFunction {
       }
       const end = time();
       const res = await loader(args);
+      const isRouterV7 = process.env._MODERN_ROUTER_VERSION === 'v7';
+      if (isRouterV7) {
+        let activeDeferreds = null;
+        if (typeof document === 'undefined') {
+          activeDeferreds = getAsyncLocalStorage()?.useContext()
+            ?.activeDeferreds as Map<string, DeferredData>;
+        } else {
+          activeDeferreds = originalActiveDeferreds;
+        }
+        if (
+          res !== null &&
+          typeof res === 'object' &&
+          Object.getPrototypeOf(res) === Object.prototype
```

Comment:
What is the meaning of this line, could you add some comment ?

Reply:
This ensures that only pure objects will be processed, I move this part into a function.